nsv13/game/lazylift.dm

```diff
@@ -0,0 +1,394 @@
+	set waitfor = FALSE
+	. = ..()
+	if(speaker == src || world.time < next_voice_activation || master.in_use)
+		return
+	if(!get_language_holder()?.has_language(message_language))
+		say("[pick("I'm sorry, I didn't quite understand that!", "Please speak clearly into the microphone.", "Unable to process voice command")]")
+		return
+	if(findtext(raw_message, "?")) //Don't ask questions, just clearly state your intent.
+		return
+	next_voice_activation = world.time + 1 SECONDS //To avoid spamming list iteration.
+	for(var/obj/machinery/lazylift/LL in master.decks)
+		if(findtext(raw_message, "deck")) //Alright they're asking for a specific deck.
+			if(findtext(raw_message, num2text(LL.deck)))
+				master.path_to(LL.deck)
+				playsound(src.loc, 'sound/machines/chime.ogg', 100, TRUE)
+				return TRUE
+		for(var/area/AR in LL.addresses)
+			var/department = AR.name
+			if(findtext(raw_message, department))
+				master.path_to(LL.deck)
+				playsound(src.loc, 'sound/machines/chime.ogg', 100, TRUE)
+				return TRUE
+
+/obj/machinery/lazylift/proc/setup()
+	set waitfor = FALSE //Don't wanna get hung up on creating doors do we?
+	var/turf/open/door_turf = null
+	for(var/turf/T in platform)
+		if(istype(T, /turf/open/openspace) || T.z != z)
+			continue
+		door_turf = T
+		var/obj/machinery/door/airlock/door = locate(/obj/machinery/door/airlock) in T
+		if(door)
+			doors += door
+	if(!door_turf)
+		message_admins("Couldn't find a turbolift door turf for [src]!")
+		return
+
+/obj/machinery/lazylift/proc/check_mobs_in_lift()
+	var/mobs = 0
+	for(var/atom/movable/AM in get_area(src))
+		if(AM.z != z)
+			continue
+		if(isliving(AM))
+			mobs ++
+	return (mobs <= 0)
+
+//Mappers, use me! The other one is generated by this thing.
+/obj/machinery/lazylift/master
+	name = "Turbolift control panel"
+	var/list/decks = list() //Lifts that are slaved to this 'un. This abuses pointers to tell a lift object to point to a given deck.
+	var/obj/machinery/lazylift/platform_location = null //Where is the lift platform at?
+	var/in_use = FALSE
+
+/obj/machinery/lazylift/Initialize()
+	. = ..()
+	for(var/turf/T in get_area(src))
+		if(T.z != z)
+			continue
+		if(!istype(T, /turf/open))
+			var/obj/machinery/lazylift_button/BB = locate(/obj/machinery/lazylift_button) in T
+			if(BB)
+				BB.lift = src
+			continue
+		else
+			platform += T
+	addtimer(CALLBACK(src, .proc/acquire_destinations), 10 SECONDS)
+	setup()
+
+/obj/machinery/lazylift/proc/acquire_destinations()
+	for(var/x in SSmapping.areas_in_z["[z]"])
+		var/area/AR = x
+		if(AR.type in area_blacklist)
+			continue
+		addresses += AR
+
+/obj/machinery/lazylift/master/Initialize()
+	. = ..()
+	master = src
+	var/turf/T = get_turf(src)
+	turf_type = T.type
+	//Time to set up the wacky turbolift network!
+	platform_location = src //IMPORTANT! Platform starts at the bottom lift.
+	var/turf/last = get_turf(src)
+	decks += src //Ew. But necessary.
+	for(var/I = 0; I <= world.maxz; I++)
```

Comment:
Have you checked to see if you can use a version of get z's of this type? Aka station_z's. This seems like a slightly inefficient method personally.

Reply:
To be honest this isn't that bad. It terminates as soon as it stops finding relevent turbolift bits. In terms of CPU operations this is:
get value of world.maxz
increment a singular number until it reaches this value, or we stop finding openspace turfs